dali/kernels/tensor_shape.h

```diff
@@ -768,6 +809,9 @@ bool operator==(const TensorListShape<left_ndim> &left, const TensorListShape<ri
   if (left.sample_dim() != right.sample_dim()) {
```

Comment:
I wonder if we should allow comparison of two fixed-sample-dim TensorListShapes. Maybe add:
```
detail::check_compatible_ndim<left_ndim, right_ndim>();
```

Reply:
Done